src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -37,649 +41,723 @@ private static string SqlFor(LockOption lockOption)
     /// or null if no matching Document was found.
     /// </summary>
-    public static async Task<JsonNode?> FindDocumentEdfiDocByDocumentUuid(
+    public async Task<JsonNode?> FindDocumentEdfiDocByDocumentUuid(
         DocumentUuid documentUuid,
         string resourceName,
         PartitionKey partitionKey,
         NpgsqlConnection connection,
         NpgsqlTransaction transaction,
-        LockOption lockOption
+        LockOption lockOption,
+        TraceId traceId
     )
     {
-        var result = await GetPostgresExceptionRetryPipeline().ExecuteAsync(async _ =>
-        {
-            try
+        var result = await GetPostgresExceptionRetryPipeline()
+            .ExecuteAsync(async _ =>
             {
-                await using NpgsqlCommand command =
-                new(
-                    $@"SELECT EdfiDoc FROM dms.Document WHERE DocumentPartitionKey = $1 AND DocumentUuid = $2 AND ResourceName = $3 {SqlFor(lockOption)};",
-                    connection,
-                    transaction
-                )
+                try
                 {
-                    Parameters =
+                    await using NpgsqlCommand command =
+                        new(
+                            $@"SELECT EdfiDoc FROM dms.Document WHERE DocumentPartitionKey = $1 AND DocumentUuid = $2 AND ResourceName = $3 {SqlFor(lockOption)};",
+                            connection,
+                            transaction
+                        )
+                        {
+                            Parameters =
+                            {
+                                new() { Value = partitionKey.Value },
+                                new() { Value = documentUuid.Value },
+                                new() { Value = resourceName },
+                            },
+                        };
+
+                    await command.PrepareAsync();
+                    await using NpgsqlDataReader reader = await command.ExecuteReaderAsync();
+
+                    if (!reader.HasRows)
                     {
-                        new() { Value = partitionKey.Value },
-                        new() { Value = documentUuid.Value },
-                        new() { Value = resourceName }
+                        return null;
                     }
-                };
 
-                await command.PrepareAsync();
-                await using NpgsqlDataReader reader = await command.ExecuteReaderAsync();
-
-                if (!reader.HasRows)
-                {
-                    return null;
+                    // Assumes only one row returned
+                    await reader.ReadAsync();
+                    return (await reader.GetFieldValueAsync<JsonElement>(0)).Deserialize<JsonNode>();
                 }
+                catch (PostgresException pe)
+                {
+                    _logger.LogError(pe, "DB failure, will retry - {TraceId}", traceId);
```

Comment:
Lack of exception logging before retry was hiding problems.

Reply:
Since this is going to be retried, I wonder if this should be a _warning_ instead of an _error_.

Reply:
```suggestion
                    _logger.LogWarning(pe, "DB failure, will retry - {TraceId}", traceId);
```